### ISSUE TITLE
Offset at search documentation/implementation bug

### ISSUE THREAD

**Issue by rmmjohann** [spotify/web-api#645]
#### Endpoint(s):
* `GET /v1/search/`

#### Problem:
`offset` is documented to accept values until 100k, but the search for playlists with specific arguments contains about 2m results. With the offset limit of 100k it's never possible to request them.

In addition, you are even not possible to query an offset of more than 10k, as already reported at #634

**Comment by hughrawlinson** [comment]
Hey @rmmjohann - we'll have a look at #634, but it seems like once that's resolved, offset will be working as documented? I'm very interested in your use case for retrieving 2m playlists?

(please reopen this issue if you feel like I've missed something)

**Comment by rmmjohann** [comment]
Use case is analytics on playlists.

Don't get me wrong: It's weird that this issue has been closed without leaving any comment on issue on #634.

Since reporting this issue the search API has been changed: Before I was able to search for playlists with q=*, now the response is
```json
{
  "error" : {
    "status" : 400,
    "message" : "No search query"
  }
}
```
What's the reason for this change?

It would be great if we are able to access every playlist by requesting with q=* and traversing the result set with offsets across 10,000.
